### ISSUE TITLE
Unnecessary parentheses generated in ConditionAsField

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10304]
Some `Condition` implementations never require any parentheses e.g. when wrapped in `ConditionAsField`. For example:

```java
select(field(exists(select(inline(1)))))
select(field(not(exists(select(inline(1))))))
```

This currently produces:

```sql
select (exists (select 1))
select (not (exists (select 1)))
```

But it could produce, instead:

```sql
select exists (select 1)
select not exists (select 1)
```

See also: https://github.com/jOOQ/jOOQ/issues/9554 (`not(unique(...))` vs `notUnique()`)

**Comment by lukaseder** [comment]
The parentheses were added because of https://github.com/jOOQ/jOOQ/issues/4506

**Comment by lukaseder** [comment]
There are a few other parentheses related issues, such as e.g. https://github.com/jOOQ/jOOQ/issues/7362. Better investigate this topic in a more global context - there are always syntactic caveats in SQL, where suddenly, parentheses are needed.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0
